packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -326,6 +326,18 @@ class IosProject extends XcodeBasedProject {
     await _updateGeneratedXcodeConfigIfNeeded();
   }
 
+  String? xcworkspacePath() {
```

Comment:
Return the `Directory?` instead of a `String`.

Actually looking at this again, this method already exists, so that's the spot that should be updated:

https://github.com/flutter/flutter/blob/9f5c6553bcbd53cc012869504a9c9d89c4133577/packages/flutter_tools/lib/src/xcode_project.dart#L50

That would also correct these, which are based off the `xcodeWorkspace` property.
https://github.com/flutter/flutter/blob/9f5c6553bcbd53cc012869504a9c9d89c4133577/packages/flutter_tools/lib/src/xcode_project.dart#L53-L56

Reply:
So just to make sure I understand. Instead of using `_hostAppProjectName`, which gets hardcoded as `Runner`, in the `xcodeWorkspace` getter do the loop logic to determine the xcworkspace?

Reply:
If yes, the `xcodeWorkspace` getter return a not-nullable Directory. Should I return like a fallback if the loop doesn't find it? Perhaps what it was originally `hostAppRoot.childDirectory('$_hostAppProjectName.xcworkspace');`?

Or do you think I should change it to return a nullable Directory and try and refactor the code that uses it? Maybe throw an error if it's not found?

In my opinion, a fallback doesn't really make sense because if the loop doesn't find anything, then there isn't a Runner.xcworkspace. But on the other hand, that's what it was originally.

Reply:
I wouldn't just throw an error here since it's up to the callsite what it needs the workspace for.  For example, in `XcodeBuildSystemMigration` it's being used to check if it exists, and if it doesn't it just skips the migration, and shouldn't error.

I glanced at the places it's used and I don't think it would be too difficult to change `xcodeWorkspace` to a nullable `Directory`.  Give it a shot and see if it makes sense.

Reply:
Updated in https://github.com/flutter/flutter/pull/114099/commits/1596b41ec707513c5f56150bfe088076b71be469.